sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/RecognizePiiEntityAsyncClient.java

```diff
@@ -0,0 +1,205 @@
+                .map(warning -> {
+                    final WarningCodeValue warningCodeValue = warning.getCode();
+                    return new TextAnalyticsWarning(
+                        WarningCode.fromString(warningCodeValue == null ? null : warningCodeValue.toString()),
+                        warning.getMessage());
+                }).collect(Collectors.toList());
+
+            recognizeEntitiesResults.add(new RecognizePiiEntitiesResult(
+                documentEntities.getId(),
+                documentEntities.getStatistics() == null ? null
+                    : toTextDocumentStatistics(documentEntities.getStatistics()),
+                null,
+                new PiiEntityCollection(new IterableStream<>(piiEntities), new IterableStream<>(warnings))
+            ));
+        });
+        // Document errors
+        entitiesResult.getErrors().forEach(documentError -> {
+            recognizeEntitiesResults.add(
+                new RecognizePiiEntitiesResult(documentError.getId(), null,
+                    toTextAnalyticsError(documentError.getError()), null));
+        });
+
+        return new SimpleResponse<>(response,
+            new RecognizePiiEntitiesResultCollection(recognizeEntitiesResults, entitiesResult.getModelVersion(),
+                entitiesResult.getStatistics() == null ? null : toBatchStatistics(entitiesResult.getStatistics())));
+    }
+
+    /**
+     * Call the service with REST response, convert to a {@link Mono} of {@link Response} that contains
+     * {@link RecognizePiiEntitiesResultCollection} from a {@link SimpleResponse} of {@link EntitiesResult}.
+     *
+     * @param documents The list of documents to recognize Personally Identifiable Information entities for.
+     * @param options The {@link TextAnalyticsRequestOptions} request options.
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     *
+     * @return A mono {@link Response} that contains {@link RecognizePiiEntitiesResultCollection}.
+     */
+    private Mono<Response<RecognizePiiEntitiesResultCollection>> getRecognizePiiEntitiesResponse(
+        Iterable<TextDocumentInput> documents, TextAnalyticsRequestOptions options, Context context) {
+        return service.entitiesRecognitionPiiWithResponseAsync(
+            new MultiLanguageBatchInput().setDocuments(toMultiLanguageInput(documents)),
+            options == null ? null : options.getModelVersion(),
+            options == null ? null : options.isIncludeStatistics(),
+            null,
+            context.addData(AZ_TRACING_NAMESPACE_KEY, COGNITIVE_TRACING_NAMESPACE_VALUE))
+            .doOnSubscribe(ignoredValue -> logger.info("A batch of documents - {}", documents.toString()))
+            .doOnSuccess(response ->
+                logger.info("Recognized Personally Identifiable Information entities for a batch of documents- {}",
+                response.getValue()))
```

Comment:
Should it be fine to log PII information here?

Reply:
Good point, i can hide them.